attribute/set.go

```diff
@@ -29,14 +29,19 @@ type (
 	// This type supports the Equivalent method of comparison using values of
 	// type Distinct.
 	Set struct {
-		equivalent Distinct
+		// id is the key to the sets registry where this set's data is.
+		//
+		// A pointer is used so the finalizer can handle reference-counting for
+		// the sets registry while still being optimized as a map key.
+		id *uint64
```

Comment:
Would it be possible to make this work by using a pointer to `setData` instead of a pointer to a separate `uint64`?